### PR TITLE
Fix blocksizes in ELPA

### PR DESCRIPTION
**Pull request by fstein93**
This PR enforces the block size to be a power of 2 (also on CPU) if an eigenvalue decomposition with ELPA is requested.

### CODE REVIEW

File: src/fm/cp_fm_elpa.F [hunk 1]
```diff
@@ -75,33 +81,50 @@ MODULE cp_fm_elpa
...
+   INTEGER, DIMENSION(18 &
+#if defined (__ELPA_NVIDIA_GPU)
+                      + 1 &
+#endif
+#if defined (__ELPA_AMD_GPU)
+                      + 1 &
+#endif
+#if defined (__ELPA_INTEL_GPU)
+                      + 1 &
+#endif
```
Comment (oschuett) [inline src/fm/cp_fm_elpa.F]: Why is this needed? AFAIK, those constants are always available - even when the GPU kernels are not.

**Comment by oschuett** [comment]
It would be awesome if this fixes #464!

File: src/fm/cp_fm_elpa.F (hunk 1 above)
Comment (fstein93) [reply src/fm/cp_fm_elpa.F]: On our cluster, ELPA_2STAGE_REAL_INTEL_GPU_SYCL is not available. Or, at least, I receive compiler errors due the missing symbol ELPA_2STAGE_REAL_INTEL_GPU_SYCL in elpa_constants.  The other two are explicitly checked later in the code. That's why I added the respective checks, too.
The issue is that you have to provide the length of fixed-size array explicitly. In that fashion, I can circumvent compilation issues because the compiler can obviously determine this size at compile-time.

Comment (oschuett) [reply src/fm/cp_fm_elpa.F]: The `ELPA_2STAGE_REAL_INTEL_GPU_SYCL` kernel was added only recently. So, you are probably using an older version of ELPA. Given that ELPA had a number of severe issues in the past, I'm not sure we should actually support backwards compatibility.

Comment (fstein93) [reply src/fm/cp_fm_elpa.F]: Then, we can actually remove these kind of checks.

Comment (fstein93) [reply src/fm/cp_fm_elpa.F]: I removed the respective commit.

**Comment by fstein93** [comment]
I have rerun some of the failing calculations on Daint and obtained the correct results. Do you think it is reasonable to employ Scalapack instead of ELPA in all failing tests? I know for sure that some passing tests (mostly from RPA/MP2) definitely make use of eigenvalue decompositions.

**Comment by oschuett** [comment]
>  Do you think it is reasonable to employ Scalapack instead of ELPA in all failing tests?

No, we should really get to the bottom of this. Otherwise I'd have very little trust in the results - despite some tests occasionally passing. Fortunately, most of the tests failed because ELPA itself returned an error code. So, those should be rather straightforward to investigate.

**Comment by oschuett** [comment]
The timeouts are probably due to https://github.com/marekandreas/elpa/issues/17.
I guess, until this point we never actually used ELPA's GPU kernels.

**Comment by fstein93** [comment]
Apart from two special cases which require results by other regtests and the timeouts, there is only one kind of error in the bandred code of ELPA. I will consult the ELPA developers for that. It might take some time.

**Comment by fstein93** [comment]
Apparently, the error is related to the 2stage solver because we observe timeouts if we employ the 1stage solver instead.

**Comment by oschuett** [comment]
It seems #2447 fixed the timeouts. @fstein93, could you please revert back to the two-stage solver?

**Comment by fstein93** [comment]
Apparently, everything works now. If there are no further issues raised, I will merge this PR.

**Comment by oschuett** [comment]
I'm a bit disappointed that we don't see any improvements in our [performance test](https://dashboard.cp2k.org/archive/perf-cuda-volta/index.html).
Shouldn't this show up in the `diag_cu144_broy.inp` benchmark?

**Comment by fstein93** [comment]
Might be but I don't know what kernel we had used before.
